app/chains/itinerary_chain.py

```diff
@@ -84,12 +95,80 @@ def extract_chat_history_content(x):
     chat_history=extract_chat_history_content
 ) | prompt | llm
 
+session_memories = {}
+
+def _cleanup_expired_sessions():
+    """
+    Remove session histories that have not been accessed within TTL to avoid memory leaks.
+    No-op if TTL is non-positive.
+    """
+    try:
+        ttl = SESSION_MEMORY_TTL_SECONDS
+        now = time.time()
+        expired_session_ids = []
+        for s_id, entry in list(session_memories.items()):
+            if not isinstance(entry, dict) or "last_access" not in entry:
+                continue
+            if now - entry["last_access"] > ttl:
+                expired_session_ids.append(s_id)
+        for s_id in expired_session_ids:
+            del session_memories[s_id]
```

Comment:
Deleting session entries during iteration could cause runtime errors if the dictionary changes size. Iterate over a static list of keys to avoid mutation during iteration.
```
for s_id in list(expired_session_ids):
    del session_memories[s_id]
```